### DISCUSSION TITLE
"Don't use shared examples"

### DISCUSSION THREAD

**Discussion by glaucocustodio** [evenbetterspecs/evenbetterspecs.github.io#16] [General]
What are your thoughts on https://evenbetterspecs.github.io/#dont-use-shared-examples?

**Comment by asavageiv** [comment]
How donyou ensure a set of classes conform to an interface? What if I add a method to the interface?

**Reply by glaucocustodio** [reply]
I would have a test for each class, even because all of them would have different implementations..

**Reply by v-yarotsky** [reply]
Having a test for each class cannot guarantee that both classes conform to the same interface.
Example from real life: 2 interchangeable data storage mechanisms for a service – postgres and dynamodb. I had 2 implementations, and one shared examples group included in their respective specs. Worked beautifully!
